source/databricks/calculation_engine/package/calculation/preparation/transformations/charges_reader.py

```diff
@@ -11,17 +11,45 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from datetime import datetime
 
 from pyspark.sql import DataFrame
+
+from package.calculation.preparation.transformations.clamp_period import clamp_period
 from package.constants import Colname
 from package.calculation_input import TableReader
 
 
-def read_charges(table_reader: TableReader) -> DataFrame:
-    charge_master_data_df = table_reader.read_charge_master_data_periods()
-    charge_links_df = table_reader.read_charge_links_periods()
-    charge_prices_df = table_reader.read_charge_price_points()
+def read_charges(
+    table_reader: TableReader,
+    period_start_datetime: datetime,
+    period_end_datetime: datetime,
+) -> DataFrame:
+    charge_master_data_df = table_reader.read_charge_master_data_periods(
+        period_start_datetime, period_end_datetime
+    )
+    charge_master_data_df = clamp_period(
```

Comment:
clamp? How about this
```suggestion
    charge_master_data_df = filter_by_period(
```

Reply:
it's not really a filter. It sets to/from date to the specified limits if they exceed.